test/soca/CMakeLists.txt

```diff
@@ -12,3 +12,108 @@ ecbuild_add_test(
         OBS_DIR=${OBS_DIR}
         PYTHONPATH=${PROJECT_BINARY_DIR}/ush:$ENV{PYTHONPATH}
 )
+
+# Prepare backgrounds
+# -------------------
+# 1 - create fake 3 hourly restarts under RESTART
+# 2 - store them with R2D2
+set( TESTDATA ${PROJECT_BINARY_DIR}/test/testdata )
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bkg)
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bkg/RESTART)
+ecbuild_add_test(
+      TARGET    test_gdasapp_soca_bkgdb
+      TYPE      SCRIPT
+      COMMAND   ${Python3_EXECUTABLE}
+      ARGS      ${PROJECT_SOURCE_DIR}/test/soca/create_bkgdb.py
+      WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bkg
+      ENVIRONMENT
+        TESTDATA=${TESTDATA}
+        PYTHONPATH=${PROJECT_BINARY_DIR}/ush:$ENV{PYTHONPATH}
+)
+
+# Symlink test input yaml files
+# -----------------------------
+# create testinput dir
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/testinput)
+
+# list of test input files to install
+list(APPEND test_input
+  ${PROJECT_SOURCE_DIR}/parm/soca/obs/obs_list.yaml
+  ${PROJECT_SOURCE_DIR}/test/soca/testinput/dumy.yaml
+)
+
+# symlink
+foreach(FILENAME ${test_input})
+  get_filename_component(filename ${FILENAME} NAME )
+  execute_process( COMMAND ${CMAKE_COMMAND} -E create_symlink
+           ${FILENAME}
+           ${PROJECT_BINARY_DIR}/test/soca/testinput/${filename} )
+endforeach(FILENAME)
+
+# install
+install(FILES ${test_input}
+        DESTINATION "test/testinput/")
+
+# Set runtime environement variables
+# ----------------------------------
+set( CDATE 2018041512 )        # Center of current cycle date
```

Comment:
Rather than all this cmake, why not just make the test run a shell script that has all of this defined in it? Wouldn't that be cleaner? What I do is just pass like `${CMAKE_CURRENT_BINARY_DIR}` to the shell script as `$1` and then define everything else inside the script. See https://github.com/NOAA-EMC/GDASApp/blob/develop/test/atm/test_convert_gsi_satbias.sh

Reply:
will do